### PR TITLE
Statically link the macos release with libPNG and SDL2_Net

### LINKED ISSUE #416: IPX support unavailable in macOS build
This seems to be an odd thing to leave out since it's built into the official Dosbox. I'd love to be able to play some network games. Any chance you'll be putting this back in?

### PR DESCRIPTION
**Pull request by kcgen**
Adds `--enable-png-static` to link PNG statically, and now `--enable-sdl-static` will also link SDL2_Net statically.

On macos:
- The `.a` libraries are assumed to be in the expected Brew-provided `/usr/local/lib/` directory.

    ``` text
     ccache clang++ -std=gnu++14  -DNDEBUG -O3 -fno-math-errno -fstrict-aliasing -march=nehalem -flto=thin -pipe -mno-ms-bitfields    -o dosbox dosbox.o  cpu/libcpu.a debug/libdebug.a dos/libdos.a fpu/libfpu.a gui/libgui.a hardware/libhardware.a hardware/mame/libmame.a hardware/serialport/libserial.a ints/libints.a libs/decoders/libdecoders.a libs/gui_tk/libgui_tk.a libs/nuked/libnuked.a libs/ppscale/libppscale.a midi/libmidi.a misc/libmisc.a shell/libshell.a /usr/local/lib/libSDL2.a -lm -liconv -Wl,-framework,CoreAudio -Wl,-framework,AudioToolbox -Wl,-framework,ForceFeedback -lobjc -Wl,-framework,CoreVideo -Wl,-framework,Cocoa -Wl,-framework,Carbon -Wl,-framework,IOKit -Wl,-weak_framework,QuartzCore -Wl,-weak_framework,Metal -framework CoreMIDI -framework CoreFoundation -framework AudioUnit -framework AudioToolbox /usr/local/lib/libpng.a -lz /usr/local/lib/libSDL2_net.a /Users/runner/runners/2.263.0/work/dosbox-staging/dosbox-staging/contrib/static-opus/lib/libopusfile.a /Users/runner/runners/2.263.0/work/dosbox-staging/dosbox-staging/contrib/static-opus/lib/libogg.a /Users/runner/runners/2.263.0/work/dosbox-staging/dosbox-staging/contrib/static-opus/lib/libopus.a -lm -framework OpenGL -framework CoreFoundation
    ```

    ``` text
    2020-06-11T15:15:07.5413610Z + otool -L src/dosbox
    2020-06-11T15:15:08.5911110Z src/dosbox:
    2020-06-11T15:15:08.5911440Z 	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0,                 current version 1281.100.1)
    2020-06-11T15:15:08.5911730Z 	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
    2020-06-11T15:15:08.5911870Z 	/System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio (compatibility version 1.0.0, current version 1.0.0)
    2020-06-11T15:15:08.5913150Z 	/System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox (compatibility version 1.0.0, current version 1000.0.0)
    2020-06-11T15:15:08.5913480Z 	/System/Library/Frameworks/ForceFeedback.framework/Versions/A/ForceFeedback (compatibility version 1.0.0, current version 1.0.2)
    2020-06-11T15:15:08.5913620Z 	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
    2020-06-11T15:15:08.5913750Z 	/System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo (compatibility version 1.2.0, current version 1.5.0)
    2020-06-11T15:15:08.5913870Z 	/System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa (compatibility version 1.0.0, current version 23.0.0)
    2020-06-11T15:15:08.5914010Z 	/System/Library/Frameworks/Carbon.framework/Versions/A/Carbon (compatibility version 2.0.0, current version 162.0.0)
    2020-06-11T15:15:08.5914140Z 	/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
    2020-06-11T15:15:08.5914290Z 	/System/Library/Frameworks/QuartzCore.framework/Versions/A/QuartzCore (compatibility version 1.2.0, current version 1.11.0, weak)
    2020-06-11T15:15:08.5914430Z 	/System/Library/Frameworks/Metal.framework/Versions/A/Metal (compatibility version 1.0.0, current version 212.5.15, weak)
    2020-06-11T15:15:08.5914560Z 	/System/Library/Frameworks/CoreMIDI.framework/Versions/A/CoreMIDI (compatibility version 1.0.0, current version 69.0.0)
    2020-06-11T15:15:08.5914700Z 	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1675.129.0)
    2020-06-11T15:15:08.5914830Z 	/System/Library/Frameworks/AudioUnit.framework/Versions/A/AudioUnit (compatibility version 1.0.0, current version 1.0.0)
    2020-06-11T15:15:08.5914960Z 	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
    2020-06-11T15:15:08.5915090Z 	/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)
    2020-06-11T15:15:08.5915230Z 	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 902.1.0)
    2020-06-11T15:15:08.5916560Z 	/System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 1894.40.150)
    2020-06-11T15:15:08.5916830Z 	/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 1355.13.0)
    2020-06-11T15:15:08.5916980Z 	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 1069.22.0)
    2020-06-11T15:15:08.5917120Z 	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1675.129.0)
    ```

On non-macos:

- The `.a` libs may be found dynamically by your compiler if they're in its search path(s).  Alternatively, the user can specify their own directory using `LDFLAGS="-L/path/to/static/libs/"`, which is expected behavior (and works for dynamic libraries as well).

    ``` text
    g++-9 -std=gnu++14  -O3 -mtune=native -pipe -Wnon-virtual-dtor -Weffc++ -Woverloaded-virtual -Wuseless-cast -mno-ms-bitfields   -O3 -mtune=native -pipe -o dosbox dosbox.o  cpu/libcpu.a debug/libdebug.a dos/libdos.a fpu/libfpu.a gui/libgui.a hardware/libhardware.a hardware/mame/libmame.a hardware/serialport/libserial.a ints/libints.a libs/decoders/libdecoders.a libs/gui_tk/libgui_tk.a libs/nuked/libnuked.a libs/ppscale/libppscale.a midi/libmidi.a misc/libmisc.a shell/libshell.a -lasound -lm -ldl -lpthread -lSDL2 -Wl,--no-undefined -lm -ldl -lpulse-simple -lpulse -lX11 -lXext -lXcursor -lXinerama -lXi -lXrandr -lXss -lXxf86vm -lwayland-egl -lwayland-client -lwayland-cursor -lxkbcommon -lpthread -lrt -Wl,-Bstatic -lpng -Wl,-Bdynamic -lz -Wl,-Bstatic -lSDL2_net -Wl,-Bdynamic -lopusfile -lGL
    ```

    ``` text
    ldd src/dosbox | grep -i 'libSDL\|libpng'
    libSDL2-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0 (0x00007f340cc65000)
    ```

This PR also adds these static flags to the Config Heavy workflow.

Fixes #416 

Thank you to @acegary for reporting and testing this PR, and to @dreamer for the suggestion to use static libraries.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -466,12 +473,28 @@ AC_ARG_ENABLE(network,
...
+AC_CHECK_HEADER(SDL_net.h, have_sdl_net_h=yes,)
 AC_CHECK_LIB(SDL2_net, SDLNet_Init, have_sdl_net_lib=yes, , )
 AC_MSG_CHECKING([whether networking features will be enabled])
 if test x$enable_network = xyes ; then
   if test x$have_sdl_net_lib = xyes -a x$have_sdl_net_h = xyes ; then
-    LIBS="$LIBS -lSDL2_net"
+    if test x$enable_sdl_net_static = xyes ; then
+      case "$host" in
+        *-*-darwin*)
+          LIBS="$LIBS /usr/local/lib/libSDL2_net.a"
```
Comment (dreamer) [inline configure.ac:489]: *Just a side note (not blocking this review).*

We haven't actually tested if `-Wl,-Bstatic -l<libname> <libflags> -Wl,-Bdynamic` works on macOS with brew packages! When --enable-sdl-static was being implemented, I didn't knew about `-Wl,-Bstatic -Wl,-Bdynamic` option). Perhaps we can get rid of having separate case for Darwin.

File: configure.ac [hunk 2]
```diff
@@ -466,12 +473,28 @@ AC_ARG_ENABLE(network,
...
+          ;;
+        *)
+          LIBS="$LIBS -Wl,-Bstatic -lSDL2_net -Wl,-Bdynamic"
```
Comment (dreamer) [inline configure.ac:492]:
```suggestion
          LIBS="$LIBS -Wl,-Bstatic -lSDL2_net -lSDL2 -Wl,--no-undefined -lm -ldl -lpthread -lrt -Wl,-Bdynamic"
```
As these are flags suggested by `pkg-config --static --libs SDL2_net`.

Yes, this means that `--enable-sdl-net-static` implies `--enable-sdl-static` in practice. Maybe there's a better way to handle it. I am also ok with leaving it as it is for now - if it works on macOS.

**Comment by kcgen** [comment]
> opengl is disabled

Egads!  I unfortunately can't reproduce it; OpenGL is unaffected on my side (Ubuntu 20.04):

**./configure --enable-sdl-net-static**:

``` text
checking SDL_net.h usability... yes
checking SDL_net.h presence... yes
checking for SDL_net.h... yes
checking for SDLNet_Init in -lSDL2_net... yes
checking whether networking features will be enabled... yes
checking for OPUSFILE... yes
checking for main in -lGL... yes
checking for main in -lopengl32... no
checking GL/gl.h usability... yes
checking GL/gl.h presence... yes
checking for GL/gl.h... yes
checking whether opengl display output will be enabled... yes

grep GL config.h
#define C_OPENGL 1
```

**./configure --enable-png-static**

``` text
checking png.h usability... yes
checking png.h presence... yes
checking for png.h... yes
checking for png_get_io_ptr in -lpng... yes
checking whether screenshots will be enabled... yes
checking GL/gl.h usability... yes
checking GL/gl.h presence... yes
checking for GL/gl.h... yes
checking whether opengl display output will be enabled... yes

grep GL config.h
#define C_OPENGL 1
```

**./configure --enable-sdl-net-static --enable-png-static**

``` text
checking png.h usability... yes
checking png.h presence... yes
checking for png.h... yes
checking for png_get_io_ptr in -lpng... yes
checking whether screenshots will be enabled... yes

checking for SDL_net.h... yes
checking for SDLNet_Init in -lSDL2_net... yes
checking whether networking features will be enabled... yes

checking for main in -lGL... yes
checking for main in -lopengl32... no
checking GL/gl.h usability... yes
checking GL/gl.h presence... yes
checking for GL/gl.h... yes
checking whether opengl display output will be enabled... yes

grep GL config.h
#define C_OPENGL 1
```

**./configure --enable-sdl-net-static --enable-opengl**

``` text
checking SDL_net.h usability... yes
checking SDL_net.h presence... yes
checking for SDL_net.h... yes
checking for SDLNet_Init in -lSDL2_net... yes
checking whether networking features will be enabled... yes

checking for main in -lGL... yes
checking for main in -lopengl32... no
checking GL/gl.h usability... yes
checking GL/gl.h presence... yes
checking for GL/gl.h... yes
checking whether opengl display output will be enabled... yes

grep GL config.h
#define C_OPENGL 1
```

File: configure.ac (hunk 2 above)
Comment (kcgen) [reply configure.ac:491]: I think there's a better way: `--enable-sdl-static` should throw that lever for SDL in general (and now we finally have the underlying mechanism to honor it for SDL2_Net as well).  Let me try again.

Comment (kcgen) [reply configure.ac:491]: **./configure --enable-png-static**

``` text
checking png.h usability... yes
checking png.h presence... yes
checking for png.h... yes
checking for png_get_io_ptr in -lpng... yes
checking whether screenshots will be enabled... yes (static)
```

**./configure  --enable-sdl-static**

``` text
checking SDL_net.h usability... yes
checking SDL_net.h presence... yes
checking for SDL_net.h... yes
checking for SDLNet_Init in -lSDL2_net... yes
checking whether networking features will be enabled... yes (static)
```

That's looking better than the original commit, and saner behavior with SDL being static overall or not, and only having to throw one switch.  If someone finds a scenario were we need to tease these apart again, then it we can look into splitting the SDL switches again.  Although hopefully by then, we'll be on cmake and it can be smart enough to fallback to dynamic libs whenever a static one isn't found :-)

**Comment by kcgen** [comment]
Last round of commits passed for the [macOS release build](https://pipelines.actions.githubusercontent.com/RDKYnb9yFkXPfcXINdFWH3mzDaNqlgzrXM4kV8rnx6PJBbLLEF/_apis/pipelines/1/runs/11678/signedlogcontent/6?urlExpires=2020-06-12T15%3A01%3A57.3521787Z&urlSigningMethod=HMACV1&urlSignature=uBXwuBr2uNDFowOvatf0PCN6CouHYrSkNr9LW%2B2iTdA%3D). 

**./configure --enable-png-static --enable-sdl-static** 

``` text
2020-06-12T14:56:33.6642050Z checking png.h usability... yes
2020-06-12T14:56:33.7181540Z checking png.h presence... yes
2020-06-12T14:56:33.7183990Z checking for png.h... yes
2020-06-12T14:56:33.8871980Z checking for png_get_io_ptr in -lpng... yes
2020-06-12T14:56:33.8879230Z checking whether screenshots will be enabled... yes (static)
2020-06-12T14:56:34.4494360Z checking SDL_net.h usability... yes
2020-06-12T14:56:34.5753450Z checking SDL_net.h presence... yes
2020-06-12T14:56:34.5754690Z checking for SDL_net.h... yes
2020-06-12T14:56:34.7381690Z checking for SDLNet_Init in -lSDL2_net... yes
2020-06-12T14:56:34.7392940Z checking whether networking features will be enabled... yes (static)
```

The binary is free from `/usr/local/` dynamic links:

**otool -L src/dosbox**

``` text
2020-06-12T14:58:02.8304800Z + strip src/dosbox
2020-06-12T14:58:02.8649630Z + otool -L src/dosbox
2020-06-12T14:58:03.7113550Z src/dosbox:
2020-06-12T14:58:03.7113900Z 	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
2020-06-12T14:58:03.7114040Z 	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
2020-06-12T14:58:03.7114180Z 	/System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio (compatibility version 1.0.0, current version 1.0.0)
2020-06-12T14:58:03.7114370Z 	/System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox (compatibility version 1.0.0, current version 1000.0.0)
2020-06-12T14:58:03.7115360Z 	/System/Library/Frameworks/ForceFeedback.framework/Versions/A/ForceFeedback (compatibility version 1.0.0, current version 1.0.2)
2020-06-12T14:58:03.7115580Z 	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
2020-06-12T14:58:03.7115720Z 	/System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo (compatibility version 1.2.0, current version 1.5.0)
2020-06-12T14:58:03.7115850Z 	/System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa (compatibility version 1.0.0, current version 23.0.0)
2020-06-12T14:58:03.7115990Z 	/System/Library/Frameworks/Carbon.framework/Versions/A/Carbon (compatibility version 2.0.0, current version 162.0.0)
2020-06-12T14:58:03.7116120Z 	/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
2020-06-12T14:58:03.7116260Z 	/System/Library/Frameworks/QuartzCore.framework/Versions/A/QuartzCore (compatibility version 1.2.0, current version 1.11.0, weak)
2020-06-12T14:58:03.7116410Z 	/System/Library/Frameworks/Metal.framework/Versions/A/Metal (compatibility version 1.0.0, current version 212.5.15, weak)
2020-06-12T14:58:03.7116550Z 	/System/Library/Frameworks/CoreMIDI.framework/Versions/A/CoreMIDI (compatibility version 1.0.0, current version 69.0.0)
2020-06-12T14:58:03.7116700Z 	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1675.129.0)
2020-06-12T14:58:03.7116840Z 	/System/Library/Frameworks/AudioUnit.framework/Versions/A/AudioUnit (compatibility version 1.0.0, current version 1.0.0)
2020-06-12T14:58:03.7116970Z 	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
2020-06-12T14:58:03.7117100Z 	/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)
2020-06-12T14:58:03.7117590Z 	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 902.1.0)
2020-06-12T14:58:03.7117740Z 	/System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 1894.40.150)
2020-06-12T14:58:03.7117880Z 	/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 1355.13.0)
2020-06-12T14:58:03.7118020Z 	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 1069.22.0)
2020-06-12T14:58:03.7118160Z 	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1675.129.0)
2020-06-12T14:58:03.7174150Z ##[group]Run set -x
```

**Comment by dreamer** [comment]
```
$ ./configure --enable-png-static --enable-sdl-static
(…)
checking png.h usability... yes
checking png.h presence... yes
checking for png.h... yes
checking for png_get_io_ptr in -lpng... yes
checking whether screenshots will be enabled... yes (static)
checking SDL_net.h usability... yes
checking SDL_net.h presence... yes
checking for SDL_net.h... yes
checking for SDLNet_Init in -lSDL2_net... yes
checking whether networking features will be enabled... yes (static)
checking for OPUSFILE... yes
checking for main in -lGL... no
checking for main in -lopengl32... no
checking GL/gl.h usability... yes
checking GL/gl.h presence... yes
checking for GL/gl.h... yes
checking whether opengl display output will be enabled... no
```

The output I normally get is (highlighted only difference and opengl parts):
```
$ ./configure
(…)
checking whether screenshots will be enabled... yes 
(…)
checking whether networking features will be enabled... yes 
(…)
checking for main in -lGL... yes             # <- with static sdl it says no!
checking for main in -lopengl32... no
checking GL/gl.h usability... yes
checking GL/gl.h presence... yes
checking for GL/gl.h... yes
checking whether opengl display output will be enabled... yes
```

Let me look, maybe it's something that could be easily mitigated.

**Comment by dreamer** [comment]
OK, I think I found the problem:

- Fedora does not provide `libSDL2_net.a` file - there's only `libSDL2.a` static lib, there are no static packages for SDL2_net at all. SDL2_net is not bundled in that single static lib.
- Despite of this my configure still says:
```
checking SDL_net.h usability... yes
checking SDL_net.h presence... yes
checking for SDL_net.h... yes
checking for SDLNet_Init in -lSDL2_net... yes
checking whether networking features will be enabled... yes (static)
```
(But then fails during linking stage, as static lib is not found by linker)

When above happens, updated LIBS env variable *probably* quietly leaks into later checks, specifically into:
```
AC_CHECK_LIB(GL, main, have_gl_lib=yes, have_gl_lib=no , )
```
Due to missing SDL2_net static, this check fails (as binary compiled by autoconf internally to check if GL is available fails), thus marking OpenGL as not available (which is false).

*I hate autoconf.*

I just checked, and removing SDL2_net headers (to disable support) and compiling with `--enable-sdl-static` does not work on my Linux either (SDL2 is linked dynamically; but OpenGL is detected fine, because `am_path_sdl.m4` uses SDL_LIB env variable internally, so bad flags are not leaked to other checks.

### Conclusion

Static linking of SDL2 does not work reliably, at least on some Linuxes, with our configure.ac script as written right now. I don't really care about it, as in practice we need static linking only for macOS anyway.

It's just N-th issue with autoconf we're facing - I *think* the proper solution would be to move all SDL2 checks into a single m4 file (and radically simplify them), but it would be a lot of work and, in my opinion, **it's not worth the trouble**.

@kcgen How about we just add AC_MSG_WARN when using static linking outside of macOS? On your Linux, with all libs available it worked fine, so there's a case to leaving the code as it is right now.

File: configure.ac [hunk 3]
```diff
@@ -466,23 +475,36 @@ AC_ARG_ENABLE(network,
...
-    LIBS="$LIBS -lSDL2_net"
+    libtype=""
+    if test x$enable_sdl_static = xyes ; then
+      libtype="(static)"
+      case "$host" in
+        *-*-darwin*)
+          LIBS="$LIBS /usr/local/lib/libSDL2_net.a"
+          ;;
+        *)
+          LIBS="$LIBS -Wl,-Bstatic -lSDL2_net -Wl,-Bdynamic"
```
Comment (dreamer) [inline configure.ac:491]: Suggestion: before line 491:

    AC_MSG_WARN([Static linking of SDL2 does not work reliably. Make sure you have libSDL2_net.a file available.])

Comment (kcgen) [reply configure.ac:494]: Added.

For PNG, I was able to replace the old style handling with PKG_CONFIG (and specifying a minimum version, which is nice); but unfortunately this trick didn't work for SDL2_Net because it's output include -lSDL2 (in addition to -lSDL2_net), and wrapping this statically: `libsdl2net_LIBS="-Wl,-Bstatic $libsdl2net_LIBS -Wl,-Bdynamic"` causes subsequent `./configure` link checks to fail - so it's a no go.

That said, after `./configure` is done, you can drop `-Wl,-Bstatic $libsdl2net_LIBS -Wl,-Bdynamic` into the Makefile, and it works no problem and produces a fully-static dosbox bin. Argh.  Autotools is a nightmare.

**Comment by kcgen** [comment]
New behavior of PNG:

**Old systems:**
``` text
 checking for libpng... yes
configure: screenshots enabled... using -lpng12
```

**Newer systems:**

``` text
checking for libpng... yes
configure: screenshots enabled... using -lpng16 -lz
```

or statically:

``` text
checking for libpng... yes
configure: screenshots enabled... using -Wl,-Bstatic -lpng16 -lz -Wl,-Bdynamic
```

macOS:

``` text
checking for libpng... yes
configure: screenshots enabled... using -L/usr/local/Cellar/libpng/1.6.37/lib -lpng16 -lz
```

or static:

``` text
 configure: checking for libpng... using macOS brew library
configure: screenshots enabled... using /usr/local/lib/libpng.a
```

**Comment by dreamer** [comment]
Merged-in as: e5a414cd88f79b8469bb8f4855646c939c3fe772

@kcgen Thanks! :)

**Comment by kcgen** [comment]
> Following this change, on Fedora 32 (which is arguably a modern system), --enable-png-static now causes autoconf to fail detection of both SDL_net and OpenGL

Break out the Picard face-palm. Sigh. Autotools is some kind of voluntary software-based torture device.  

I'm even drinking their Koolade using PKG_CONFIG_MODULE, and somehow checking for `libpng` breaks the checks for SDL_net and OpenGL. I don't understanding  how it gets such simple things so wrong. 

Fully agree it's not worth waste more time on; we will celebrate with a glass of cmake Koolade one day :-)
